src/sys/mman.rs

```diff
@@ -142,6 +145,9 @@ libc_bitflags! {
         /// Region grows down, like a stack.
         #[cfg(any(linux_android, freebsdlike, target_os = "openbsd"))]
         MAP_STACK;
+        /// Do not write through the page caches, write directly to the file. Used for Direct Access (DAX) enabled file systems.
+        #[cfg(all(any(target_arch = "x86"), not(linux_android), not(target_os = "hurd"), not(target_env = "uclibc"), not(freebsdlike)))]
```

Comment:
This flag is not that lucky,  uClibc does not have it, for the glibc and musl, seems that they all lack the support for the MIPS target, so this should work for now:


```suggestion
        /// Do not write through the page caches, write directly to the file. Used for Direct Access (DAX) enabled file systems.
        // Available on Linux glibc and musl, MIPS* target excluded.
        #[cfg(all(target_os = "linux", not(any(target_arch = "mips", target_arch = "mips64", target_arch = "mipsel", target_arch = "mips64el")), not(target_env = "uclibc")))]
```

Reply:
That's great, thanks for the suggestions and apologies again for all the clutter